pkg/trace/sampler/coresampler.go

```diff
@@ -74,11 +78,12 @@ type Sampler struct {
 }
 
 // newSampler returns an initialized Sampler
-func newSampler(extraRate float64, maxTPS float64) *Sampler {
+func newSampler(extraRate float64, maxTPS float64, rateThresholdTo1 float64) *Sampler {
```

Comment:
This is most of the time equal to `defaultSamplingRateThresholdTo1`. Can we maybe remove this argument and add a method:
```
func (s *Sampler) setRateThresholdTo1(r float64) *Sampler {
    s.rateThresholdTo1 = r
}
```
Then we would keep the simple constructor, and you would only need to call this method in the `PriorityEngine`:
```
s.Sampler.setRateThresholdTo1(prioritySamplingRateThresholdTo1)
```
What do you think? Seems a bit prettier.